### PR TITLE
Create a plugin module

### PR DESCRIPTION
**Pull request by Diachk**
related to https://github.com/Linkurious/linkurious-server/issues/1913

### CODE REVIEW

File: src/models/Plugins.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ * Copyright Linkurious SAS 2012 - 2019
+ *
+ * - Created on 2019-09-17.
+ */
+
+export enum PluginState {
+  RUNNING = 'running',
+  STOPPED = 'stopped',
+  ERROR_RUNTIME = 'error-runtime',
+  ERROR_CONFIGURATION = 'error-configuration',
```
Comment (francescoinfante) [inline src/models/Plugins.ts]: @Diachk let's remove error-configuration after all

File: src/models/Plugins.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+/**
+ * LINKURIOUS CONFIDENTIAL
+ * Copyright Linkurious SAS 2012 - 2019
+ *
+ * - Created on 2019-09-17.
+ */
+
```
Comment (francescoinfante) [inline src/models/Plugins.ts]: @Diachk header 
// TS2019-DONE

File: src/module/PluginsModule.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import { IInstalledPlugin } from '../models/Plugins';
+import { Unauthorized } from '../response/errors';
```
Comment (francescoinfante) [inline src/module/PluginsModule.ts]: import after the header

File: src/module/PluginsModule.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+/**
+ * LINKURIOUS CONFIDENTIAL
+ * Copyright Linkurious SAS 2012 - 2019
+ *
+ * - Created on 2019-09-17.
+ */
+import {Module} from './Module';
+
+export class PluginsModule extends Module {
```
Comment (francescoinfante) [inline src/module/PluginsModule.ts]: You have to add the module to src/index.ts files to have the module instantiated by the rest-client and the models to index.ts for getting the types from the server.

File: src/module/PluginsModule.ts [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+  /**
+   * Get the list of installed plugins.
+   */
+  public getPlugins(): Promise<Success<IInstalledPlugin[]> | Unauthorized> {
```
Comment (francescoinfante) [inline src/module/PluginsModule.ts]: Let's stick to response interfaces:
Promise<Success<IGetPluginsResponse> | ...

then you can define :
- type IGetPluginsResponse = IInstalledPlugin[]

File: src/models/Plugins.ts [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+export enum PluginState {
+  RUNNING = 'running',
+  STOPPED = 'stopped',
+  ERROR_RUNTIME = 'error-runtime',
+  ERROR_CONFIGURATION = 'error-configuration',
+  ERROR_MANIFEST = 'error-manifest'
+}
+
+export interface IInstalledPlugin {
+  pluginName: string;
```
Comment (francescoinfante) [inline src/models/Plugins.ts]: name

File: src/models/Plugins.ts (hunk 2 above)
Comment (Diachk) [reply src/models/Plugins.ts]: What does this mean in the rest client ?

File: src/models/Plugins.ts [hunk 7]
```diff
@@ -0,0 +1,22 @@
+/**
+ * LINKURIOUS CONFIDENTIAL
```
Comment (davidrapin) [inline src/models/Plugins.ts]: Files in the REST client should not be marked as confidential.
We can keep the copyright line, but "LINKURIOUS CONFIDFENTIAL" must go

Comment (francescoinfante) [reply src/models/Plugins.ts]: changing it globally

**Pull request merged**